### PR TITLE
#CP-34 CRUDs operations for Routes

### PR DESCRIPTION
**Pull request by Abdulkeza**
# What does this PR do?
-Contains the CRUDs operations for Routes

# Description of Task to be completed?
- [x] The operator should be able to create
- [x] The operator should be able to  delete
- [x] The operator should be able to update a route.

# How should this be manually tested?
[Clone](https://github.com/atlp-rwanda/phantom-fe-codebandits.git)
git co ft-crude-operations-for-routes-cp-34
npm i
npm start

# Any background context you want to provide?
(N/A)
# What are the relevant pivotal tracker stories?
[WX0cthYM](https://trello.com/c/WX0cthYM/34-crud-operations-for-routes)

### CODE REVIEW

File: src/pages/routes/components/RouteComponent.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                if (selectedRoute) {setDistance(selectedRoute[0].distance) 
+                    return
+                }
+            }
+        };
+        
+        useEffect(() => {
+            distanceValue(destination1, destination2);
+        }, [destination2, destination2]);
+
```
Comment (Yvad60) [inline src/pages/routes/components/RouteComponent.js]: Make sure that the form validation is working as expected so far when you match destination 1 and destination 2 you get the error but when you choose destination 2 first validation doesn't work and a route without distance will be registered. and that's the same case with updating routes

File: src/pages/routes/components/RouteComponent.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        
+        <div className = "cursor-not-allowed rounded-sm p-2 mb-5 bg-[#EFEFEF] disabled:" >
+             { distance || `${data.distance }`}</div> 
+        <div className = "self-center mt-6" >
+        <Button name = {formTitle} styles = "bg-primary text-sm text-white py-2 px-3 hover:bg-hover" />
+        </div> 
+        </form >
+        </div>
+          )
+    };
```
Comment (Yvad60) [inline src/pages/routes/components/RouteComponent.js]: add validation for this form currently when you just click register button a route without data will be registered

**Review by Yvad60 — CHANGES_REQUESTED** [review]
Good work on the routes CRUD Mr. Abdul. take a look at the following feedback

File: src/pages/routes/components/RouteComponent.js (hunk 2 above)
Comment (Abdulkeza) [reply src/pages/routes/components/RouteComponent.js]: - [x] ## **Well** Solved

**Review by Yvad60 — DISMISSED** [review]

**Review by samyshyaka — APPROVED** [review]
This is great work Abdul

**Review by feyton — APPROVED** [review]
Requested features are all implemented

**Pull request merged**